intermine_boot/intermine_docker.py

```diff
@@ -0,0 +1,308 @@
+
+    return postgres_container
+
+
+def create_intermine_builder_container(client, image, env):
+    user = _get_docker_user()
+
+    data_dir = env['data_dir'] / 'docker' / 'data'
+
+    environment = {
+        'MINE_NAME': os.environ.get('MINE_NAME', 'biotestmine'),
+        'MINE_REPO_URL': os.environ.get('MINE_REPO_URL', ''),
+        'MEM_OPTS': os.environ.get('MEM_OPTS', '-Xmx2g -Xms1g'),
+        'IM_DATA_DIR': os.environ.get('IM_DATA_DIR', ''),
+        'IM_REPO_URL': os.environ.get('IM_REPO_URL', ''),
+        'IM_REPO_BRANCH': os.environ.get('IM_REPO_BRANCH', '')
+    }
+
+    mine_path = env['data_dir'] / 'docker' / 'data' / 'mine'
+
+    volumes = {
+        mine_path / 'dump': {
+            'bind': '/home/intermine/intermine/dump',
+            'mode': 'rw'
+        },
+
+        mine_path / 'configs': {
+            'bind': '/home/intermine/intermine/configs',
+            'mode': 'rw'
+        },
+        mine_path / 'packages': {
+            'bind': '/home/intermine/.m2',
+            'mode': 'rw'
+        },
+        mine_path / 'intermine': {
+            'bind': '/home/intermine/.intermine',
+            'mode': 'rw'
+        },
+        mine_path / 'biotestmine': {
+            'bind': '/home/intermine/intermine/biotestmine',
+            'mode': 'rw'
+        }
+    }
+
+    print ('\n\nStarting Intermine container...\n\n')
+
+    try:
+        assert client.containers.get('postgres').status == 'running'
+    except AssertionError:
+        print ('Postgres container not running. Exiting...')
+        exit(1)
+
+    try:
+        assert client.containers.get('tomcat').status == 'running'
+    except AssertionError:
+        print ('Tomcat container not running. Exiting...')
+        exit(1)
+
+    try:
+        assert client.containers.get('solr').status == 'running'
+    except AssertionError:
+        print ('Solr container not running. Exiting...')
+
+    intermine_builder_container = _start_container(
+        client, image, name='intermine_builder', user=user, environment=environment,
+        volumes=volumes, network=DOCKER_NETWORK_NAME)
+
+    return intermine_builder_container
+
+
+def _start_container(client, image, name, user=None, environment=None, volumes=None, network=None, ports=None, log_match=None):
+    try:
+        container = client.containers.run(
+            image, name=name, user=user, environment=environment,
+            volumes=volumes, network=network, detach=True)
+        
+        for log in container.logs(stream=True, timestamps=True):
+            print (log)
+            if log_match is not None and log_match in str(log):
+                break
+    except docker.errors.ImageNotFound as e:
+        print ('docker image not found for %s ' % container_name, e.msg)
+        exit(1)
+    except docker.errors.ContainerError as e:
+        print ('Error while running container ', e.msg)
+        exit(1)
+    
+    return container
```

Comment:
**Very** clean and smart! Great job :100:

Reply:
Thanks!